File: javascript/060.js
//  060
//  Permutation Sequence
//  2015-12-16
/////////////////////////////////////////////////////
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    var s = "";
    var f = 1;
    var d = [];
    for (var i = 1; i <= n; i++) {
        f *= i;
        d.push(i);
    }
    for (i = n; i > 0; i--) {
        f /= i;
        var q = Math.floor((k-1) / f);
        s += d[q];
        d.splice(q, 1);
        k -= q*f;
    }
    return s;
};

/////////////////////////////////////////////////////
var n = 3;
var k = 4;
console.log(getPermutation(n, k));

File: python/060.py
#   060
#   Permutation Sequence
#   2015-12-16
#####################################################
import math
class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        s = ""
        f = math.factorial(n)
        d = [str(i) for i in range(1, n+1)]
        for i in range(n, 0, -1):
            f //= i
            q = (k-1) // f
            s += d[q]
            del d[q]
            k -= q*f
        return s 

#####################################################
sol = Solution()

n = 3
k = 4

z = sol.getPermutation(n, k)
print(z)

